sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -753,23 +702,14 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
                         cancellationToken).ConfigureAwait(false);
                     if (message == null)
                     {
-                        // if we are processing next available session, and no messages were returned:
-                        // 1. cancel the renew session lock task
-                        // 2. break out of the loop to allow requesting another session from the service
-                        if (IsSessionProcessor && _sessionIds.Length == 0)
+                        // If we are doing session processing, break out of the loop to allow a new session to
+                        // be processed.
+                        if (IsSessionProcessor)
```

Comment:
Don't you have to reopen tostarte getting new messages?

Reply:
Yes, but this is handled automatically by RunReceiveTaskAsync